### DISCUSSION TITLE
Joomla 3.9.23 when changing a template then custom menutypes are lost

### DISCUSSION THREAD

**Discussion by B3nito** [joomla/joomla-cms#31548] [Q&A]
### Steps to reproduce the issue
If you use a template with custom menutypes for example another type of blog with xml and you switch the template then the menutype is just empty and lost. 

### Expected result
The item should use the custom menu type of other template or custom menutypes should store different and not in template. It not makes sense that functionality of site is broken.

### Actual result
Broken site. Even if new template has exact same menutypes

**Comment by ReLater** [comment]
> then the menutype is just empty and lost.

I can't confirm that. As long as the template is not deleted other templates can use it. Or am I wrong?

The drama starts when you delete that template plus overrides.

As a workaround I use a folder template/shared-menutypes/ with just one subfolder /html/ with my menu type overrides in it that I want to share with several other templates.

But yes, I also would prefer a more handy solution in core.
